### PR TITLE
gnuradio: add new package with dependencies 

### PR DESCRIPTION
**Pull request by chrisjohgorman**
I am quite new to msys2, but I decided in order to learn how to
package for msys2, that I would build gnuradio under msys2 and offer
up the PKGBUILD files. I have done this now and I am ready to release
this to everyone else. I would consider the software advanced alpha
or new beta in terms of issues.

# Issues
1. Not built and tested under clangarm64. I haven't got the hardware
and didn't try to figure out how to cross compile for clangarm64. If
someone has pointers on how to do this, I would really appreciate
them. The packages all build and run on mingw64, ucrt64 and clang64.
2. Currently I'm using the git master branch of gnuradio, I would
prefer to move to a release, but needed to use the master branch for a
small patch. I also had a dependency error with qwt-qt5 needing to be
below 6.3.0, which is the current version in mingw-w64, and this went
away when I built with the master branch.
3. I would like to eventually split gnuradio into two packages, the
libraries and docs. The docs make the file huge relative to the
gnuradio from arch which doesn't include the docs. The docs
themselves are about 366 MB.
4. I haven't tested gr_modtool.py, but will probably this afternoon.
5. When clicking on running the gr_filter_design.py file, Windows
asks what file to use when opening the gr_filter_design.py. One has
to use the python.exe file in ${MINGW_PREFIX}/bin/.
6. In order to use soapysdr and soapyrtlsdr packages, with gnuradio,
they need to be rebuilt. I'm not sure what the issue is here, but the soapy
gnuradio block doesn't work unless both packages are rebuilt on my machine.
7. Probably others.

# To build
There are only ten packages, so the build order shouldn't be too hard.
This is the order I use.

mingw-w64-castxml
mingw-w64-python-pygccxml
mingw-w64-codec2
mingw-w64-libiio
mingw-w64-libad9361-iio
mingw-w64-libuhd
mingw-w64-mathjax2
mingw-w64-python-click-plugins
mingw-w64-python-pyaml
mingw-w64-gnuradio

### CODE REVIEW

**Comment by chrisjohgorman** [comment]
I have fixed two missing dependencies for python-click-plugins.  (makedepends - python-setuptools and checkdepends - python-click).  This hopefully will fix the build.

**Comment by chrisjohgorman** [comment]
I watched the build and noticed an error in the `gnuradio` package.  It is missing a build dependency on `doxygen`.  It still built fine, but the docs won't be as complete.   When you're looking at this PR, could you tell me how to go about adding the dependency now that it has passed all checks?  Do I wait for the packages to be released, then make a change to the gnuradio package, or can I put the change in now?

Thanks

Chris

**Comment by mmuetzel** [comment]
You can update the PR any time before it is merged.
If you'd like to take your time in polishing it, you could also convert it to draft until it is ready as an indicator for reviewers.

**Comment by chrisjohgorman** [comment]
Thanks Markus.  There should only be the need for this fix at the moment.

File: mingw-w64-gnuradio/PKGBUILD [hunk 1]
```diff
@@ -4,7 +4,7 @@ _realname=gnuradio
 pkgbase=mingw-w64-${_realname}
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
 pkgver=3.11.0.0git.r790.g0491c2d5e
-pkgrel=1
+pkgrel=2
```
Comment (mmuetzel) [inline mingw-w64-gnuradio/PKGBUILD:7]: You don't need to bump the pkgrel for changes within the same PR. IIUC, it's actually preferred to not do that.

**Comment by chrisjohgorman** [comment]
Thanks again Markus.  I have corrected it.

Chris

**Comment by mmuetzel** [comment]
@chrisjohgorman: One of the last steps for getting a PR merged is testing whether the build packages work (if that is possible).

You can download the build artifacts from the CI and install the packages with `pacman -U` to test them. Do they work for you?

**Comment by chrisjohgorman** [comment]
The build artifacts don't run on my machine.  The reason for this is there is a directory mismatch between my machine and the build machines.  When the packages are built, the python files are headered with #!D:\M\msys64\mingw64\bin\python3.exe.  This won't run on a machine that doesn't have python.exe at D:\M\msys64\mingw64\bin\.  (My machine has python3.exe at c:\msys64\mingw64\bin).

Additionally my sed scripts to setup the path for the blocks path and examples path have the same difficulty.  The block/examples path is setup with the build machine's root (d:\M\...) and not the root on my machine (C:\...).

I will look into this and see what I can do to fix it, but if anyone has ideas on how to rectify this, I would appreciate them.

Thanks

Chris

**Comment by mmuetzel** [comment]
> The build artifacts don't run on my machine. The reason for this is there is a directory mismatch between my machine and the build machines. When the packages are built, the python files are headered with #!D:\M\msys64\mingw64\bin\python3.exe. This won't run on a machine that doesn't have python.exe at D:\M\msys64\mingw64\bin. (My machine has python3.exe at c:\msys64\mingw64\bin).

IIUC, the original shebang (`#!/usr/bin/env python`, e.g., [here](https://github.com/gnuradio/gnuradio/blob/b404dd7e325af9888e497922f5dc6b970e454f33/grc/scripts/gnuradio-companion#L1)) should be working in MSYS2. 
Does it run correctly if you replace the line with the absolute path with the original line from the repository?

In case, the original shebang is working:
Afaict, that line is replaced here:
https://github.com/gnuradio/gnuradio/blob/b404dd7e325af9888e497922f5dc6b970e454f33/cmake/Modules/GrPython.cmake#L355-L362

Maybe, you could skip that replacement when packaging for MSYS2?

> Additionally my sed scripts to setup the path for the blocks path and examples path have the same difficulty. The block/examples path is setup with the build machine's root (d:\M...) and not the root on my machine (C:...).

Which sed scripts are you referring to? What do they do? And what issue are they intended to fix?

**Comment by chrisjohgorman** [comment]
Thanks for the help.  The original shebang (#!/usr/bin/env python) does work in msys2, but I need to change at least one value in the file ${MINGW_PREFIX}/etc/gnuradio/conf.d/grc.conf to point the global_blocks_path variable to the appropriate windows directory.  Currently, I'm doing this via some sed substitutions in the gnuradio PKGBUILD file.  (This is what I referred to as sed scripts.)  If I do this substitution in the PKGBUILD then it comes out with the build machine's path.  What I want is to have the global_blocks_path point to the appropriate directory on the install machine.  If this path is not set properly, gnuradio-companion.py doesn't load.   Specifically, if a user with the msys2 root D:\msys64, I set the global_blocks_path in the grc.conf to D:\msys64\${MINGW_PREFIX}\share\gnuradio\grc\blocks.  So I guess if I want this to happen I need to run a post install script via alpm-hooks.  Since I need to do this for global_blocks_path, I might as well do this to fix the shebangs and examples_path at the same time.

I have a rough outline of a script to run.  
```
#!/bin/bash

#DEPENDS find, which, msys2-runtime, sed, bash

python_path=$(which python3.exe);
windows_python_path=$(cygpath.exe -w ${python_path});
escaped_windows_python_path=$(printf '%s\n' "${windows_python_path}" | sed -e 's/[]\/$*.^[]/\\&/g');

list_of_files_to_change="bin/gnuradio-companion.py bin/gr-ctrlport-monitor.py bin/gr-perf-monitorx.py bin/grcc.py \
 bin/gr_filter_design.py bin/gr_modtool.py bin/gr_plot.py bin/gr_plot_const.py bin/gr_plot_fft.py bin/gr_plot_iq.py \
 bin/gr_plot_psd.py bin/gr_plot_qt.py bin/gr_read_file_metadata.py bin/polar_channel_construction.py bin/uhd_fft.py \
 bin/uhd_rx_cfile.py bin/uhd_rx_nogui.py bin/uhd_siggen.py bin/uhd_siggen_gui.py"

#reset shebang to local system layout
for file in ${list_of_files_to_change}
do
   sed -i "1s/.*/#\!${escaped_windows_python_path}/" "${MINGW_PREFIX}/${file}"
done

#Invert slashes from back slash to forward slash
sed -i 's/\\/\//g' "${MINGW_PREFIX}/etc/gnuradio/conf.d/grc.conf"

#Make global_blocks_path use windows directory naming using cygpath
gbp=$(grep global_blocks_path "${MINGW_PREFIX}/etc/gnuradio/conf.d/grc.conf");
gbpv=$(printf '%s\n' "${gbp}" | cut -d = -f2);
replace_gbpv=$(cygpath.exe -w $gbpv);
escaped_replace_gbpv=$(printf '%s\n' "${replace_gbpv}" | sed -e 's/[]\/$*.^[]/\\&/g');
sed -i 's@'"${gbpv}"'@'\ ${escaped_replace_gbpv}'@' "${MINGW_PREFIX}/etc/gnuradio/conf.d/grc.conf"

#Make examples_path use windows directory naming using cygpath
ep=$(grep examples_path "${MINGW_PREFIX}/etc/gnuradio/conf.d/grc.conf");
epv=$(printf '%s\n' "${ep}" | cut -d = -f2);
replace_epv=$(cygpath.exe -w $epv);
escaped_replace_epv=$(printf '%s\n' "${replace_epv}" | sed -e 's/[]\/$*.^[]/\\&/g');
sed -i 's@'"${epv}"'@'\ ${escaped_replace_epv}'@' "${MINGW_PREFIX}/etc/gnuradio/conf.d/grc.conf"
```
I would like to have this script modified to have list_of_files_to_change populated automatically from a list generated by a find call run after install in the PKGBUILD file.   Then I have this script run by a pacman hook.

Something like ...
```
[Trigger]
Type = Package
Operation = Install
Operation = Upgrade
Target = "gnuradio"

[Action]
Description = Updating gnuradio python shebangs, global_blocks_path and examples_path ...
When = PostTransaction
Exec = /usr/share/libalpm/scripts/"script-to-run"
```
Any ideas on my strategy and whether or not it will work would be appreciated.

**Comment by chrisjohgorman** [comment]
Ok.  I've thought about this and am now convinced that the correct way to fix this is what @mmuetzel suggested.  I will still need to make a post install hook and run a shell script on install to set the `global_blocks_path` variable, but I will change it so that it's the only thing it is run for.

**Comment by chrisjohgorman** [comment]
@mmuetzel I was wondering if it is too late to change the pkgver of the gnuradio package.  I only ask now as I am currently using the development branch of gnuradio.  I think the development branch would work well, but I worry that I will not be able to "upgrade" to the next release as the version number of the next release will most likely be less than the current pkgver.  (My current pkgver is 3.11.0.0git.r790.g0491c2d5e and the latest release is 3.10.10.0.)  I believe I can get a working build with a patched 3.10.10.0, but I haven't made the necessary changes yet to use 3.10.10.0.  

Otherwise, I believe I now have a working build with using the default shebang (#!/usr/bin/env python) and the new post install hook for the default_blocks_path.

**Comment by mmuetzel** [comment]
It's still possible to change to a released version.
Testers that installed the build artifacts of this PR would need to downgrade manually once this is merged. But that shouldn't be an issue.

**Comment by chrisjohgorman** [comment]
Tested artifacts mingw64, ucrt64 and clang64 on my machine and they look good.  I am able to install and run with the artifacts.

File: mingw-w64-gnuradio/0007-remove-full-windows-path-on-shebang.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/cmake/Modules/GrPython.cmake b/cmake/Modules/GrPython.cmake
+index dbea5b05e..5672a3152 100644
+--- a/cmake/Modules/GrPython.cmake
++++ b/cmake/Modules/GrPython.cmake
+@@ -357,7 +357,7 @@ function(GR_PYTHON_INSTALL)
+                 DEPENDS ${pyfile}
+                 COMMAND
+                     ${PYTHON_EXECUTABLE} -c
+-                    "import re; R=re.compile('^\#!.*$\\n',flags=re.MULTILINE); open(r'${pyexefile}','w',encoding='utf-8').write(r'\#!${pyexe_native}'+'\\n'+R.sub('',open(r'${pyfile}','r',encoding='utf-8').read()))"
++                    "import re; R=re.compile('^\#!.*$\\n',flags=re.MULTILINE); open(r'${pyexefile}','w',encoding='utf-8').write(open('${pyfile}',encoding='utf-8').read())"
```
Comment (mmuetzel) [inline mingw-w64-gnuradio/0007-remove-full-windows-path-on-shebang.patch:10]: Afaict, the regular expression isn't used any more. Does the following also work:
``` suggestion
+                    "open(r'${pyexefile}','w',encoding='utf-8').write(open('${pyfile}',encoding='utf-8').read())"
```
If it does, isn't that just a simple copy of the original file? Can that be written simpler than that?


IIUC, the following a couple of lines further down in the same file adds the file extension `.py` to these files:
https://github.com/gnuradio/gnuradio/blob/ac8c60c89c412d5f640b1c41cade0877aec94b64/cmake/Modules/GrPython.cmake#L366-L368
```
            if(WIN32 AND NOT pyfile_ext)
                set(pyfile_name "${pyfile_name}.py")
            endif()
```

To we really need that? Would it make sense to patch that to be the following?
```
            if(MSVC AND NOT pyfile_ext)
                set(pyfile_name "${pyfile_name}.py")
            endif()
```

Comment (mmuetzel) [reply mingw-w64-gnuradio/0007-remove-full-windows-path-on-shebang.patch:10]: Would that mean that you could drop `0004-Add-.py-extension-for-gr_filter_design-so-it-will-la.patch`?

**Comment by mmuetzel** [comment]
Perusing the build logs, I see the following (only checked MINGW64 and CLANG64):
```
  [MSYS2 CI] mingw-w64-codec2: Runtime dependencies for mingw-w64-x86_64-codec2
  
  /mingw64/bin/libcodec2.dll:
  	libgcc_s_seh-1.dll => not found
```
`codec2` should probably have a runtime dependency on `gcc-libs`.

Additionally:
```
  [MSYS2 CI] mingw-w64-libiio: Runtime dependencies for mingw-w64-x86_64-libiio
  
  /mingw64/bin/iio_attr.exe:
  	libwinpthread-1.dll => not found
  	libiio.dll => D:\M\msys64\mingw64\bin\libiio.dll (0x0000024a9ac90000)
  	 libusb-1.0.dll => not found
  	 libxml2-2.dll => not found
  /mingw64/bin/iio_genxml.exe:
  	libwinpthread-1.dll => not found
  	libiio.dll => D:\M\msys64\mingw64\bin\libiio.dll (0x0000018215440000)
  	 libusb-1.0.dll => not found
  	 libxml2-2.dll => not found
  /mingw64/bin/iio_info.exe:
  	libwinpthread-1.dll => not found
  	libiio.dll => D:\M\msys64\mingw64\bin\libiio.dll (0x00000212e0500000)
  	 libusb-1.0.dll => not found
  	 libxml2-2.dll => not found
  /mingw64/bin/iio_readdev.exe:
  	libwinpthread-1.dll => not found
  	libiio.dll => D:\M\msys64\mingw64\bin\libiio.dll (0x0000025b1bd90000)
  	 libusb-1.0.dll => not found
  	 libxml2-2.dll => not found
  /mingw64/bin/iio_reg.exe:
  	libwinpthread-1.dll => not found
  	libiio.dll => D:\M\msys64\mingw64\bin\libiio.dll (0x000002a6b4db0000)
  	 libusb-1.0.dll => not found
  	 libxml2-2.dll => not found
  /mingw64/bin/iio_writedev.exe:
  	libwinpthread-1.dll => not found
  	libiio.dll => D:\M\msys64\mingw64\bin\libiio.dll (0x000001d324d20000)
  	 libusb-1.0.dll => not found
  	 libxml2-2.dll => not found
  /mingw64/bin/libiio.dll:
  	libusb-1.0.dll => not found
  	libxml2-2.dll => not found
```

`libiio` should probably have runtime dependencies to `libwinpthread`, `libusb`, and `libxml2`.

And:
```
  [MSYS2 CI] mingw-w64-libuhd: Runtime dependencies for mingw-w64-clang-x86_64-libuhd
  
  /clang64/bin/libuhd.dll:
  	libboost_filesystem-mt.dll => D:\M\msys64\clang64\bin\libboost_filesystem-mt.dll (0x0000023e7e240000)
  	libboost_serialization-mt.dll => D:\M\msys64\clang64\bin\libboost_serialization-mt.dll (0x0000023e7f570000)
  	libboost_thread-mt.dll => D:\M\msys64\clang64\bin\libboost_thread-mt.dll (0x0000023e7e240000)
  	libusb-1.0.dll => D:\M\msys64\clang64\bin\libusb-1.0.dll (0x0000023e7e240000)
  	libpython3.11.dll => not found
  	libc++.dll => D:\M\msys64\clang64\bin\libc++.dll (0x0000023e7f570000)
```
Should probably depend on `python`.

And for `gnuradio`:
```
  /clang64/bin/libgnuradio-fft.dll:
  	libgnuradio-runtime.dll => D:\M\msys64\clang64\bin\libgnuradio-runtime.dll (0x0000022d1ed40000)
  	 libboost_thread-mt.dll => D:\M\msys64\clang64\bin\libboost_thread-mt.dll (0x0000022d1ee90000)
  	 libboost_program_options-mt.dll => D:\M\msys64\clang64\bin\libboost_program_options-mt.dll (0x0000022d1ee90000)
  	 libgmp-10.dll => D:\M\msys64\clang64\bin\libgmp-10.dll (0x0000022d1ee90000)
  	 libunwind.dll => D:\M\msys64\clang64\bin\libunwind.dll (0x0000022d1ee90000)
  	libgnuradio-pmt.dll => D:\M\msys64\clang64\bin\libgnuradio-pmt.dll (0x0000022d1ee90000)
  	libvolk.dll => D:\M\msys64\clang64\bin\libvolk.dll (0x0000022d1ef10000)
  	 liborc-0.4-0.dll => D:\M\msys64\clang64\bin\liborc-0.4-0.dll (0x0000022d1f330000)
  	libfftw3f-3.dll => not found
  	libfftw3f_threads-3.dll => not found
  	libspdlog.dll => D:\M\msys64\clang64\bin\libspdlog.dll (0x0000022d1ee90000)
  	libfmt.dll => D:\M\msys64\clang64\bin\libfmt.dll (0x0000022d1ef00000)
  	libc++.dll => D:\M\msys64\clang64\bin\libc++.dll (0x0000022d1ef10000)
```
Which probably means that it should depend on `fftw`.

And:
```
  /clang64/bin/libgnuradio-qtgui.dll:
  	libgnuradio-filter.dll => D:\M\msys64\clang64\bin\libgnuradio-filter.dll (0x0000023f69730000)
  	Qt5Widgets.dll => not found
  	libgnuradio-fft.dll => D:\M\msys64\clang64\bin\libgnuradio-fft.dll (0x0000023f69820000)
  	 libfftw3f-3.dll => not found
  	 libfftw3f_threads-3.dll => not found
  	libgnuradio-runtime.dll => D:\M\msys64\clang64\bin\libgnuradio-runtime.dll (0x0000023f69870000)
  	 libboost_thread-mt.dll => D:\M\msys64\clang64\bin\libboost_thread-mt.dll (0x0000023f69ac0000)
  	 libboost_program_options-mt.dll => D:\M\msys64\clang64\bin\libboost_program_options-mt.dll (0x0000023f69ac0000)
  	 libgmp-10.dll => D:\M\msys64\clang64\bin\libgmp-10.dll (0x0000023f69ac0000)
  	 libunwind.dll => D:\M\msys64\clang64\bin\libunwind.dll (0x0000023f69ac0000)
  	libgnuradio-pmt.dll => D:\M\msys64\clang64\bin\libgnuradio-pmt.dll (0x0000023f69ac0000)
  	libvolk.dll => D:\M\msys64\clang64\bin\libvolk.dll (0x0000023f69b40000)
  	 liborc-0.4-0.dll => D:\M\msys64\clang64\bin\liborc-0.4-0.dll (0x0000023f69f60000)
  	qwt-qt5.dll => not found
  	Qt5Gui.dll => not found
  	Qt5Core.dll => not found
  	libspdlog.dll => D:\M\msys64\clang64\bin\libspdlog.dll (0x0000023f69ac0000)
  	libfmt.dll => D:\M\msys64\clang64\bin\libfmt.dll (0x0000023f69b30000)
  	libc++.dll => D:\M\msys64\clang64\bin\libc++.dll (0x0000023f69b40000)
```
Which probably means that it should at least have `optdepends` (if not `depends`) on `qt5-base`.

(I meant the packages with with respective prefixes everywhere.)

**Comment by chrisjohgorman** [comment]
Thank you Markus!  I will make the required changes.

File: mingw-w64-libad9361-iio/PKGBUILD [hunk 3]
```diff
@@ -12,7 +12,9 @@ url='https://github.com/analogdevicesinc/libad9361-iio'
 license=(
   LGPL-2.1-or-later
 )
-depends=("${MINGW_PACKAGE_PREFIX}-libiio")
+depends=("${MINGW_PACKAGE_PREFIX}-libiio"
+         "${MINGW_PACKAGE_PREFIX}-libxml2"
+         "${MINGW_PACKAGE_PREFIX}-libusb")
```
Comment (mmuetzel) [inline mingw-w64-libad9361-iio/PKGBUILD:17]: IIUC, these are only indirect dependencies (through `libiio`):
```
  [MSYS2 CI] mingw-w64-libad9361-iio: Runtime dependencies for mingw-w64-x86_64-libad9361-iio
  
  /mingw64/bin/libad9361.dll:
  	libiio.dll => D:\M\msys64\mingw64\bin\libiio.dll (0x000001f181130000)
  	 libusb-1.0.dll => not found
  	 libxml2-2.dll => not found
```

So, it shouldn't be necessary to add them here again.

Comment (chrisjohgorman) [reply mingw-w64-libad9361-iio/PKGBUILD:17]: Thanks again for all the help.  I'll fix it and push it now.

**Comment by chrisjohgorman** [comment]
@mmuetzel I missed a commit for gnuradio and as such it is missing two dependencies, so I will have to run my build again.  I apologize to your poor build machine.  Before I run again, I wanted to know your opinion on turning this into a draft again.  Upstream gnuradio is about to make a new release and I will want to update the PKGBUILD to use the new tar ball.  When it's released in say a week or so.  Should I just wait until then, with this as a draft or release this and then in a week or so release the changes?

**Comment by mmuetzel** [comment]
The runners in the CI are hosted by GitHub. They are probably always on anyway. And I guess they are pretty efficient because they are sharing load with multiple VMs running on each machine. I personally wouldn't worry too much about build cycles...

When it comes to setting the PR to draft or not, that is up to you. If you prefer that this won't get merged before the next upstream release, setting it to draft would be a good indicator for reviewers or maintainers that this PR shouldn't be merged yet.

**Comment by chrisjohgorman** [comment]
Well, I am kind of excited to get it released and I don't mind doing the update in a week or so, so I'll push to release this, once it's ready to go.  Then when the next release is ready, I'll make the required changes.  Thanks for the mentoring during this process.  You've been very gracious with your time.

File: mingw-w64-gnuradio/0007-remove-full-windows-path-on-shebang.patch (hunk 2 above)
Comment (mmuetzel) [reply mingw-w64-gnuradio/0007-remove-full-windows-path-on-shebang.patch:10]: This was marked as resolved. But afaict, the regular expression is still there.
Did removing it cause issues for you?

**Comment by chrisjohgorman** [comment]
I'm terribly sorry about this.  I got sidetracked and forgot to write
all this down in the logs.  This particular patch
(0007-remove-full-windows-path-on-shebang.patch) is no longer being
used and should have been removed as should patch
0004-Add-.py-extension-for-gr_filter_design-so-it-will-la.patch.  I
went and got permission from Ryan Volz to use his patch
(0005-cmake-Install-python-wrapper-exe-for-scripts-on-Wind.patch) and
then got it working with msys2 gnuradio.  The only caveat is that it
requires that the binaries are not stripped.  It uses ScriptMaker to
make windows executables from the python files.  I thought this was a
better solution to the problem of executing the python files.  With
this patch we get gr_filter_design.exe launching from
gnuradio-companion.exe as it does under Linux, which was my main
motivation for using patches 0004 and 0007.  I'll remove these patches
now.  Since I am removing unused patches, I can remove patch
0001-gnuradio-runtime_dll_ws2_32.patch at this time too as when it is
used with apache thrift, it causes gnuradio to be unusable.  Finally
if we wait for the update from upstream, I can remove
0002-Using-os.path.join-is-unnecessary-as-get_cofig_file_.patch and
0003-Removing-the-use-of-os.path.join-when-getting-the-pa.patch as
their changes have been merged into upstream's maint-3.10 branch in
preparation for the release of gnuradio 3.10.11.0.

**Comment by mmuetzel** [comment]
Thank you for clarifying and for your continued work to improve the packaging rules.

Sorry for asking again: Is this ready to be merged? Or do you prefer to wait for gnuradio 3.10.11.0? (I thought I understood that you preferred not to wait. But I'm no longer sure that is still the case reading your last comment.)

**Comment by chrisjohgorman** [comment]
I believe I can wait a week or so, so I will change this to a draft
again and make changes when 3.10.11.0 is released.  It will mean
changing from git to tarball and I can make a check on PGP keys as the
arch package does.  This way we'll have only one patch too.

Thank you for all your help building this and teaching me along the
way.  Naively, I thought it would be an easy process, but the process
of building the packages shown itself to be more complex than I
thought.  I now know to use ntldd.exe to verify my dependencies before
submitting my PR!

**Comment by mmuetzel** [comment]
Note to self: This currently doesn't build for CLANG64 because `mingw-w64-clang-x86_64-python-scipy` no longer exists. Check again if gnuradio builds for that environment in case scipy can be built again with LLVM Flang 19.

**Comment by mmuetzel** [comment]
Thank you again for your contribution and the follow ups.

**Comment by chrisjohgorman** [comment]
Thank you @mmuetzel for mentoring me through this process.  I was very glad for your help.

Chris

**Pull request merged**